src/Lib/Canvas.vala

```diff
@@ -73,7 +72,49 @@ public class Akira.Lib.Canvas : Goo.Canvas {
 
     public Canvas (Akira.Window window) {
         Object (window: window);
+        // Make The canvas the drop target
+        Gtk.drag_dest_set (this, Gtk.DestDefaults.ALL, targets, Gdk.DragAction.COPY);
+        this.drag_data_received.connect(this.on_drag_data_received);
     }
+    // Making The canvas recive the dropped file
+    private const Gtk.TargetEntry[] targets = {{"text/uri-list",0,0}};
+    private void on_drag_data_received (Gdk.DragContext drag_context, int x, int y, 
+        Gtk.SelectionData data, uint info, uint time)
+    {
+    //loop through list of Files
+    const string[] ACCEPTED_TYPES = {
+        "jpeg",
+        "png",
+        "tiff",
+        "svg",
+        "xml",
+        "gif"
+    };
+    foreach(string uri in data.get_uris ()){
+        string file = uri.replace("file://","").replace("file:/","");
+        file = Uri.unescape_string (file);
+        // Creating a Glib.File using a file path
+        var img_File = GLib.File.new_for_path(file);
+        var file_img_mangager = new Akira.Lib.Managers.ImageManager(img_File,GLib.Random.int_range(0,100));
```

Comment:
This is wrong, you can add a random ID as this will create conflicts when sooner or later a duplicated ID will happen.
You should look at how we create the ImageManager in other locations and follow that implementation.